packages/debug-nodejs/src/node/debug-nodejs.ts

```diff
@@ -14,13 +14,18 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
+import { injectable } from 'inversify';
+import { DebugConfiguration } from '@theia/debug/lib/common/debug-common';
+import { DebugAdapterContribution, DebugAdapterExecutable } from '@theia/debug/lib/node/debug-model';
+
 const path = require('path');
 const packageJson = require('../../package.json');
 const debugAdapterDir = packageJson['debugAdapter']['dir'];
+const psList = require('ps-list');
 
-import { injectable } from 'inversify';
-import { DebugConfiguration } from '@theia/debug/lib/common/debug-common';
-import { DebugAdapterContribution, DebugAdapterExecutable } from '@theia/debug/lib/node/debug-model';
+const DEBUG_PORT_PATTERN = /--(inspect|debug)-port=(\d+)/;
```

Comment:
Should we only monitor `node` processes, or is it better to leave it "open" to other processes?
```ts
/node.+--(inspect|debug)-port=(\d+)/
```

Reply:
Ah but in this case you only test the cmd for the given pid, so if someone wants to parse some random program, it is up to him I guess.